acceptance/tests/resource/service/ticket_5024_systemd_enabling_masked_service.rb

```diff
@@ -0,0 +1,85 @@
+require 'puppet/acceptance/temp_file_utils'
+extend Puppet::Acceptance::TempFileUtils
+
+test_name 'Systemd masked services are unmasked before attempting to start'
+
+# This test in intended to ensure that a service which was previously marked
+# as masked and then set to enabled will first be unmasked.
+confine :to, {}, hosts.select { |host| dir_exists?(host, File.join('/', 'run', 'systemd', 'system'))
```

Comment:
i like this either way.  i'm pretty sure service_enable_linux.rb `is_sysV` will be the invert of whatever this is (in posix, outside of upstart).